src/library.js

```diff
@@ -3552,9 +3552,26 @@ LibraryManager.library = {
       var func = callback.func;
       if (typeof func === 'number') {
         if (callback.arg === undefined) {
+          // Run the wasm function ptr with signature 'v'. If no function
+          // with such signature was exported, this call does not need
+          // to be emitted (and would confuse Closure)
+#if !DYNCALLS || ('dynCall_v' in IMPLEMENTED_FUNCTIONS)
           {{{ makeDynCall('v', 'func') }}}();
+#else
+#if ASSERTIONS
+          throw 'Internal Error! Attempted to invoke wasm function pointer with signature "v", but no such functions have gotten exported!';
+#endif
```

Comment:
Would it make sense to make this logic be part of `makeDynCall` itself so all callsites can benefit?

Reply:
That will result in worse codegen, since it will need to generate a dummy function. Though this is in the large runtime path and in DYNCALLS mode, so I don't particularly mind.

Reply:
Ah I see.  LGTM either way.